tests/spec/core/highlight-spec.js

```diff
@@ -74,6 +98,76 @@ describe("Core — Highlight", () => {
     };
     const doc = await makeRSDoc(ops);
     const pre = doc.getElementById("test");
-    expect(pre.querySelectorAll("span[class^=hljs-]").length).toBe(0);
+    expect(pre.querySelectorAll("span[class*=hljs-]").length).toBe(0);
+  });
+
+  it("checks if <pre> content is wrapped in <code>", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `${makeDefaultBody()}
+        <section>
+          <pre class="js">
+            function foo() {
+              alert('foo');
+            }
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+    const pre = doc.querySelector("pre");
+    const code = pre.firstElementChild;
+    expect(code.localName).toBe("code");
+    expect(code.classList.contains("hljs")).toBeTruthy();
+    expect(code.classList.contains("js")).toBeTruthy();
+  });
+
+  it("asynchronously hightlights <code> elements inside <pre>", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `${makeDefaultBody()}
+        <section>
+          <pre class="example" id="first-pre">
+            <code class="js" id="test1">
+              function one(){}
+            </code>
+            function pass(){}
+            <code class="http" id="test2">
+              Header: Test1
+            </code>
+          </pre>
+          <pre id="second-pre">
+            second function(){} is not highlighted.
+            <code class="js">
+              function three(){}
+            </code>
+            <code>
+              function four(){}
+            </code>
+            <code class="http">
+              Header: Test5
+            </code>
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+
+    const firstPre = doc.getElementById("first-pre");
+    expect(firstPre.innerHTML).toContain("function pass(){}");
+    expect(firstPre.querySelectorAll("code span[class*=hljs-]")).toBeTruthy();
+    expect(firstPre.querySelector("code:nth-child(1)").textContent).toContain(
+      "function one(){}"
+    );
+
+    const secondPre = doc.getElementById("second-pre");
+    expect(secondPre.innerHTML).toContain(
+      "second function(){} is not highlighted"
+    );
+
+    const fourthCode = secondPre.querySelector("code:nth-child(2)");
+    expect(fourthCode.classList.contains("javascript")).toBeTruthy();
+    const lastCode = secondPre.querySelector("code:last-child");
+    expect(lastCode.textContent).toContain("Header: Test5");
+    expect(lastCode.classList.contains("http")).toBeTruthy();
```

Comment:
```suggestion
    expect(lastCode.classList).toContain("http");
```

Reply:
Didn't think it would be smart enough to do this.